src/drivers/rc_epgsql.erl

```diff
@@ -0,0 +1,55 @@
+-module (rc_epgsql).
+
+-behaviour (gen_server).
+
+-export ([start_link/1]).
+-export ([init/1, handle_info/2, handle_cast/2, handle_call/3, terminate/2]).
+
+start_link(Args) ->
+  gen_server:start_link({local, ?MODULE}, ?MODULE, Args, []).
+
+init(Args) ->
+  self() ! connect,
+  process_flag(trap_exit, true),
+  {ok, #{retries => 0, connection_args => Args}}.
+
+handle_info(connect, #{connection_args := ConnArgs} = State) ->
+  try
+    Host     = maps:get(host, ConnArgs, "localhost"),
+    Username = maps:get(username, ConnArgs, os:getenv("USER")),
+    Pass     = maps:get(password, ConnArgs, ""),
+    Database = maps:get(database, ConnArgs, "test_db"),
+    case maps:get(reconnection, ConnArgs, {uniform, 4000}) of
+      {uniform, Time} ->
+        {ok, Pid} = epgsql:connect(Host, Username, Pass, [{database, Database}, {timeout, Time}]),
+        lager:info("Postgres connected. ~p~n", [Pid]),
+        {noreply, State#{reconnection_type => uniform}};
+      % {backoff, From, To} ->
+      %   {ok, Pid} = epgsql:connect(Host, Username, Pass, [{database, Database}, {timeout, From}]),
+      %   lager:info("Postgres connected. ~p~n", [Pid]),
+      %   {noreply, State#{reconnection_type => backoff}};
+      Type ->
+        lager:error("Wrong reconnect_type: ~p", [Type]),
+        {stop, bad_reconnect_type, State}
+    end
+  catch
+    Ex:Err ->
+      lager:info("Catch Error - ~p:~p~n", [Ex, Err]),
+      {noreply, State}
+  end;
+
+handle_info({'EXIT', _From, Reason}, State = #{retries := Retries}) ->
+  lager:info("Error trying to connect with postgres~nRetries: ~p~nError:~p~n", [Retries, Reason]),
+  erlang:send_after(5000, self(), connect),
```

Comment:
shouldn't this 5000 come from the configured timeout?

Reply:
yes